### PR TITLE
Add auto-rendered product layout

### LINKED ISSUE #221: Add auto-rendered product layout
Add auto-rendered product layout where the product data is rendered based on [product drop](https://www.voog.com/developers/markup/ecommerce-objects/product) data.

It should look like current "Product" layout. Use new [image](https://www.voog.com/developers/markup/tags/image) or [image_data](https://www.voog.com/developers/markup/tags/image-data) tag for product cover image.

It content type must be `product`:

```json
{
  "content_type": "product",
  "component": false,
  "file": "layouts/auto_generated_product.tpl",
  "layout_name": "auto_generated_product",
  "title": "Auto-generated product"
}
```

See also buy button rendering example in [Nuuk design](https://github.com/Voog/design-nuuk/blob/master/components/buy-button.tpl).

## TODO

* [ ] Add "Auto-generated product" layout
* [ ] Check and improve "[product list](https://www.voog.com/support/online-store/product-list-content-area)" widget styles when needed.

### PR DESCRIPTION
**Pull request by PeeterPaal**
Adds auto-rendered product layout.
Also adds a browser hack to enhance image rendering in Chrome based browsers.
Closes #221

### CODE REVIEW

File: layouts/auto_generated_product.tpl [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<!DOCTYPE html>
+{% include "template-variables" %}
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:2]: Let's use white space strip here and below.

```suggestion
{%- include "template-variables" -%}
```

File: layouts/auto_generated_product.tpl [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  {% assign product_image_state = "with-image" %}
+{% endif %}
+
+<html class="{{ view_mode }} {{ language_flags_mode }} {{ language_names_mode }} {{ language_menu_mode }} {{ site_search_mode }}" lang="{{ page.language_code }}">
+  <head prefix="og: http://ogp.me/ns#">
+    {% include "template-head" with "product_page" %}
+  </head>
+
+  <body class="item-page product-page header-menu-wide">
+    {% capture bottom_content_html %}
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:16]: Let's use white space strip.
```suggestion
    {%- capture bottom_content_html %}
```

File: layouts/auto_generated_product.tpl [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            <div class="items-body pad_t-20">
+              <div class="flex-col mar_r-40">
+                <div class="content-illustrations">
+                  <div class="content-item-box {{ product_image_state }} product-content js-content-item-box">
+                    <div class="item-top">
+                      {%- if product.image != blank -%}
+                      <div class="top-inner aspect-ratio-inner">
+                        {%- assign image_class = "item-image " | append: "not-cropped" -%}
+                        {% image product.image target_width: "1280" class: image_class loading: "lazy" %}
+                      </div>
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:49]: Intent.

File: layouts/auto_generated_product.tpl [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                        <div class="content-product-description">
+                          {%- editable product.description -%}
+                        </div>
+                      {%- endif -%}
+
+                      <div class="buy-btn-content js-buy-btn-content">
+                        {% include "buy-button" %}
+                      </div>
+                      {% content bind=product %}
+
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:89]: Extra empy line.
```suggestion
```

**Review by tanelj — CHANGES_REQUESTED** [review]
Nice work, but some comments.

* [x] Left side is too small for small screens when form content area is in right side:
Screenshot 2022-02-03 at 18 08 29

**Review by tanelj — APPROVED** [review]
Nice!

File: sources/components/custom-styles/template-cs-product.scss [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    "type": "button",
+    "editor": "toggleIcon",
+    "states": {
+      "on": "700",
+      "off": "400"
+    },
+    "icon": "bold",
+    "scope": "global"
+  */
+  #{--product-description-font-weight}: 400;
```
Comment (pxska) [inline sources/components/custom-styles/template-cs-product.scss:161]: The `font-weight` value of text content areas on the page is `100` – maybe change the description's font-weight as well?

**Review by pxska — CHANGES_REQUESTED** [review]
Cool! A few comments:

- [ ] The product description needs some space on the top if the viewport is small, otherwise the text is too close to the content on the right:
Screenshot 2022-02-08 at 10 21 15

- [ ] If you change the "maximum width" value from the design editor, the width changes, but the content-right stays the same width. It should increase in width like it does on the regular product page.

**Review by pxska — APPROVED** [review]
Looks good to me!

**Review by tanelj — CHANGES_REQUESTED** [review]
Nice and works, but one issue in Safari.

* [x] "Out of stock" button text is wrapped in Safari.
Screenshot 2022-02-10 at 13 36 57

**Review by tanelj — APPROVED** [review]

**Pull request merged**